src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -26,6 +38,8 @@
 
     private final TopicRepository topicRepository;
     private final EventTypeRepository eventTypeRepository;
+    private final ValidationStrategy validationStrategy = new EventBodyMustRespectSchema();
+    private final ValidationStrategyConfiguration vsc = new ValidationStrategyConfiguration();
```

Comment:
Yes, definitely. I think it's also related to the caching problem that we discussed briefly on Friday. The optimal solution would reduce the number of objects to be instantiated in a request and also avoid hitting the database.

But by now, since we have a single validation strategy and we do hit the database, building a registry was left out of scope.